ext/fetch/lib.rs

```diff
@@ -316,12 +321,16 @@ impl FetchPermissions for deno_permissions::PermissionsContainer {
   }
 
   #[inline(always)]
-  fn check_read(
+  fn check_read<'a>(
     &mut self,
-    path: &Path,
+    path: &'a Path,
     api_name: &str,
-  ) -> Result<(), AnyError> {
-    deno_permissions::PermissionsContainer::check_read(self, path, api_name)
+  ) -> Result<Cow<'a, Path>, AnyError> {
```

Comment:
I don't understand why it needs to have different return type, can you elaborate?

Reply:
I pushed an update that adds a description to the must_use attribute. It's to mitigate TOCTOU attacks.